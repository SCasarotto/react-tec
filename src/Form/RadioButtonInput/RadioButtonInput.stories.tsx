import React from 'react';

import { Story, Meta } from '@storybook/react';

import { RadioButtonInput, RadioButtonInputProps } from './RadioButtonInput';

export default {
  title: 'Form/RadioButtonInput',
  component: RadioButtonInput,
  argTypes: {},
} as Meta;

const Template: Story<RadioButtonInputProps> = (args) => (
  <RadioButtonInput {...args} />
);

export const Default = Template.bind({});
Default.args = {
  onChange: (e) => console.log(e.target.value),
  value: 'agreeToEmailSpam',
};
