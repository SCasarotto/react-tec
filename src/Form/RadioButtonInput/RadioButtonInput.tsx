import React from 'react';

import { Input } from './styledComponents';

export interface RadioButtonInputProps
  extends React.InputHTMLAttributes<HTMLInputElement> {}
/**
 *
 * A simple checkbox component. `RadioButtonInput` is simply a styled HTML `<input type="radio">`.
 *
 * All props are passed right through to the radiobutton. Reference [HTML documentation for more information](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio).
 *
 */
export const RadioButtonInput = React.forwardRef<
  HTMLInputElement,
  RadioButtonInputProps
>((props, ref) => {
  const { className = '', ...rest } = props;
  return (
    <Input
      className={`tec-radioButtonInput ${className}`}
      ref={ref}
      {...rest}
      type='radio'
    />
  );
});
