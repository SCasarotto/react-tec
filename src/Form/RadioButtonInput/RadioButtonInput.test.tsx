import React from 'react';

import { render, waitFor } from '@testing-library/react';

import { RadioButtonInput } from './RadioButtonInput';

describe('RadioButtonInput', () => {
  // Renders Without Error
  it('matches snapshot', () => {
    const { asFragment } = render(<RadioButtonInput />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Checked
  it('renders checked undefined', () => {
    const { getByTestId } = render(
      <RadioButtonInput data-testid='myRadioButtonInput' />,
    );
    const radioButtonInput = getByTestId('myRadioButtonInput');
    expect(radioButtonInput).not.toBeChecked();
  });
  it('renders checked false', () => {
    const { getByTestId } = render(
      <RadioButtonInput
        checked={false}
        onChange={jest.fn()}
        data-testid='myRadioButtonInput'
      />,
    );
    const radioButtonInput = getByTestId('myRadioButtonInput');
    expect(radioButtonInput).not.toBeChecked();
  });
  it('renders checked true', () => {
    const { getByTestId } = render(
      <RadioButtonInput
        checked
        onChange={jest.fn()}
        data-testid='myRadioButtonInput'
      />,
    );
    const radioButtonInput = getByTestId('myRadioButtonInput');
    expect(radioButtonInput).toBeChecked();
  });

  // Disabled
  it('renders disabled undefined', () => {
    const { getByTestId } = render(
      <RadioButtonInput data-testid='myRadioButtonInput' />,
    );
    const radioButtonInput = getByTestId('myRadioButtonInput');
    expect(radioButtonInput).not.toHaveAttribute('disabled');
  });
  it('renders disabled false', () => {
    const { getByTestId } = render(
      <RadioButtonInput
        disabled={false}
        onChange={jest.fn()}
        data-testid='myRadioButtonInput'
      />,
    );
    const radioButtonInput = getByTestId('myRadioButtonInput');
    expect(radioButtonInput).not.toHaveAttribute('disabled');
  });
  it('renders disabled true', () => {
    const { getByTestId } = render(
      <RadioButtonInput
        disabled
        onChange={jest.fn()}
        data-testid='myRadioButtonInput'
      />,
    );
    const radioButtonInput = getByTestId('myRadioButtonInput');
    expect(radioButtonInput).toHaveAttribute('disabled');
  });

  // Class Name
  it('accepts custom className', () => {
    const { container } = render(<RadioButtonInput className='myClassName' />);
    expect(container.firstChild).toHaveClass('myClassName');
  });

  // Ref
  it('renders with ref', async () => {
    const ref: { current: HTMLInputElement | null } = { current: null };
    const { getByTestId } = render(
      <RadioButtonInput ref={ref} data-testid='myRadioButtonInput' />,
    );
    const radioButtonInput = getByTestId('myRadioButtonInput');
    await waitFor(() => expect(radioButtonInput).not.toHaveFocus());
    ref.current?.focus();
    await waitFor(() => expect(radioButtonInput).toHaveFocus());
  });
});
