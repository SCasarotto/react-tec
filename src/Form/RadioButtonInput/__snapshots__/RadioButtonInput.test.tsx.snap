// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtonInput matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  cursor: pointer;
  box-sizing: border-box;
  margin: 0px;
}

.c0:disabled {
  cursor: not-allowed;
}

<input
    class="c0 tec-radioButtonInput "
    type="radio"
  />
</DocumentFragment>
`;
